domain/network/service/space_test.go

```diff
@@ -90,7 +107,12 @@ func (s *spaceSuite) TestAddSpace(c *gc.C) {
 				return nil
 			})
 
-	returnedUUID, err := NewService(s.st, s.logger).AddSpace(context.Background(), "space0", network.Id("provider-id"), []string{})
+	returnedUUID, err := NewService(s.st, s.logger).AddSpace(
+		context.Background(),
+		network.SpaceInfo{
+			Name:       "space0",
+			ProviderId: "provider-id",
+		})
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(returnedUUID.String(), gc.Equals, expectedUUID)
```

Comment:
Does this test actually test that a new uuid is generated for spaces that don't have one? I guess it checks that the uuid is the same as the one going into `state.AddSpace`, but that may also be blank.

Reply:
Indeed, this test does not mock the uuid generation so we cannot test that the method is called. However this would not be very practical (we are only generating one field).
I did add a check on the test to make sure that the generated ID is not empty. The test already provides an empty ID as input, and makes sure that the state method receives a non-empty ID. This is enough IMO.